### ISSUE TITLE
Fee Estimates stuck in a loop

### ISSUE THREAD

**Issue by Varunram** [mit-dci/lit#115]
So I was trying out a couple transactions and i noticed that for some weird amount (in my case 33333 sat among others) it seemed to get stuck between two estimated fees until I stopped the node. Log attached below.

```
2017/11/23 17:53:55.415097 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.415192 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.415281 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.415428 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.415507 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.415575 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.415656 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.415737 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.415957 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416036 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416102 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416165 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416232 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416294 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416357 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416419 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416482 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416542 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416605 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416668 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416731 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416788 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416852 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.416913 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.416976 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.417039 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.417101 80 spB, est vsize 202, fee 16160
2017/11/23 17:53:55.417164 80 spB, est vsize 268, fee 21440
2017/11/23 17:53:55.417226 80 spB, est vsize 202, fee 16160
```
(continues for like 1500 lines after which I had to shut the node down)
We could have a counter attached that can be used to exit the loop after trying to find a set of utxos (1000 maybe?) so that we aren't left waiting. I'll make a PR for this after I'm done with the RBF stuff :)

**Comment by adiabat** [comment]
Huh!  Interesting problem that I haven't seen before.  Looking at the `for fee > overshoot {` loop in MaybeSend() I see how it could happen.  I can write a quick fix for this now, even though RBF code may further change it later.

**Comment by Varunram** [comment]
Yep, weird indeed. I was able to reproduce it like 3-4 times, after which I couldn't (maybe because I'd tried so many 33333 sat transactions). I have a basic fix over at #117 (I just added a counter for the loop to break at), but it could be more elaborate. Will update you on it later today.

**Comment by adiabat** [comment]
Think I fixed it with 57ab07f02c48b029ef8f6a25b6545d28d63ccc0a
The for loop in MaybeSend is gone, and only the loop in PickUtxos is used, which can run out of utxos.
I'll leave this open for a day or two but I think it's fixed now.

**Comment by adiabat** [comment]
Oh and realizing that I'm underestimating the fee in that commit because it's not adding the change output size to the estimated size.  Small error but fixable; will fix that tomorrow before closing this issue.

**Comment by Varunram** [comment]
Great, I shall close the above PR and clean up the TODOs in the new commit plus update the RBF stuff to avoid merge conflicts.

**Comment by adiabat** [comment]
Think this is fixed with b1b7b1b379027d0791dbf0dd096bd086dd889f8a so will close this.

probably lots of work left to do in this area to get it really right though.
